public/main.js

```diff
@@ -138,34 +133,72 @@ function generateTodoListElement(todo) {
         updInput.setAttribute("type", "text");
         var updConfirmButton = document.createElement("input");
         updConfirmButton.id = "upd-confirm" + todo.id;
+        updConfirmButton.classList.add("button");
         updConfirmButton.setAttribute("type", "submit");
         updConfirmButton.setAttribute("value", "Confirm");
         updForm.appendChild(updInput);
         updForm.appendChild(updConfirmButton);
         updInput.setAttribute("value", todo.title);
         todoText.parentNode.replaceChild(updForm, todoText);
         updForm.onsubmit = function() {
-            updateTodo(todo.id, updInput.value, todo.isComplete, reloadTodoList);
+            var dataToSend = JSON.stringify({
+                title: updInput.value,
+                completeState: todo.isComplete
+            });
+            makeRequest("api/todo/" + todo.id, "PUT", dataToSend, "Failed to update item.")
+                .then(reloadTodoList);
         };
     };
 
-    var completedBtn = document.createElement("button");
-    completedBtn.innerHTML = "Done";
-    completedBtn.className = "completed-btn";
-
+    //if the todo has been tagged as complete then add the complete css class to the element
+    //if the todo is uncomplete but already has the complete css class then remove it
     if (todo.isComplete) {
-        todoText.classList.add("todo-complete");
-    } else if (todoText.classList.contains("todo-complete")) {
-        todoText.classList.remove("todo-complete");
+        listItem.classList.add("todo-item-complete");
+        todoText.classList.add("todo-text-complete");
+    } else if (todoText.classList.contains("todo-text-complete")) {
```

Comment:
It does hit when a todo has been set as completed then set as uncompleted again - the todo is now uncompleted but still has the css class attached